torch/distributed/pipelining/stage.py

```diff
@@ -575,8 +582,8 @@ def backward_one_chunk(
         and a subsequent call to `backward_weight_one_chunk` is required to invoke dw_runner and complete the backward.
         """
 
-        if not full_backward:
-            assert self.dw_builder, "Must provide dw_builder to run partial backward"
+        # if not full_backward:
+        #     assert self.dw_builder, "Must provide dw_builder to run partial backward"
```

Comment:
yup, delete this, and i think there might be a comment that needs to be updated explaining this is now optional override